File: mejoregresado.py
class Estudiante:
    def __init__(self, nombre, promedio_final):
        self.nombre = nombre
        self.promedio_final = promedio_final


def main():
    print("Ingrese la cantidad de estudiantes:")
    cantidad_estudiantes = int(input())

    estudiantes = []
    for i in range(cantidad_estudiantes):
        print(f"Ingrese el nombre del estudiante {i + 1}:")
        nombre = input()
        print(f"Ingrese el promedio final del estudiante {i + 1}:")
        promedio_final = float(input())
        estudiantes.append(Estudiante(nombre, promedio_final))

    max_promedio_1 = float('-inf')
    max_promedio_2 = float('-inf')
    mejores_egresados = [None, None]

    for estudiante in estudiantes:
        if estudiante.promedio_final > max_promedio_1:
            max_promedio_2 = max_promedio_1
            max_promedio_1 = estudiante.promedio_final
            mejores_egresados[1] = mejores_egresados[0]
            mejores_egresados[0] = estudiante
        elif estudiante.promedio_final > max_promedio_2:
            max_promedio_2 = estudiante.promedio_final
            mejores_egresados[1] = estudiante

    print("\nMejores Egresados:")
    for i in range(2):
        print("Nombre:", mejores_egresados[i].nombre)
        print("Promedio Final:", mejores_egresados[i].promedio_final)
        print("-------------")


if __name__ == "__main__":
    main()

File: mejoregresado/src/main/java/Mejoregresado.java
import java.util.Scanner;
public class Mejoregresado {
    static class Estudiante {
        String nombre;
        double promedioFinal;
        Estudiante(String nombre, double promedioFinal) {
            this.nombre = nombre;
            this.promedioFinal = promedioFinal;
        }
    }
    public static void main(String[] args) {
        Scanner teclado = new Scanner(System.in);
        System.out.println("Ingrese la cantidad de estudiantes:");
        int cantidadEstudiantes = teclado.nextInt();
        Estudiante[] estudiantes = new Estudiante[cantidadEstudiantes];
        for (int i = 0; i < cantidadEstudiantes; i++) {
            System.out.println("Ingrese el nombre del estudiante " + (i + 1) + ":");
            String nombre = teclado.next();
            System.out.println("Ingrese el promedio final del estudiante " + (i + 1) + ":");
            double promedioFinal = teclado.nextDouble();
            estudiantes[i] = new Estudiante(nombre, promedioFinal);
        }
        double maxPromedio1 = Double.MIN_VALUE;
        double maxPromedio2 = Double.MIN_VALUE;
        Estudiante[] mejoresEgresados = new Estudiante[2];
        for (Estudiante estudiante : estudiantes) {
            if (estudiante.promedioFinal > maxPromedio1) {
                maxPromedio2 = maxPromedio1;
                maxPromedio1 = estudiante.promedioFinal;
                mejoresEgresados[1] = mejoresEgresados[0];
                mejoresEgresados[0] = estudiante;
            } else if (estudiante.promedioFinal > maxPromedio2) {
                maxPromedio2 = estudiante.promedioFinal;
                mejoresEgresados[1] = estudiante;
            }
        }
        System.out.println("\nMejores Egresados:");
        for (int i = 0; i < 2; i++) {
            System.out.println("Nombre: " + mejoresEgresados[i].nombre);
            System.out.println("Promedio Final: " + mejoresEgresados[i].promedioFinal);
            System.out.println("-------------");
        }
    }
}

/*
Ingrese la cantidad de estudiantes:
4
Ingrese el nombre del estudiante 1:
jose
Ingrese el promedio final del estudiante 1:
8
Ingrese el nombre del estudiante 2:
lenin
Ingrese el promedio final del estudiante 2:
10
Ingrese el nombre del estudiante 3:
alejandro
Ingrese el promedio final del estudiante 3:
9,8
Ingrese el nombre del estudiante 4:
jean
Ingrese el promedio final del estudiante 4:
10

Mejores Egresados:
Nombre: lenin
Promedio Final: 10.0
-------------
Nombre: jean
Promedio Final: 10.0
*/